src/ucs/sys/sys.c

```diff
@@ -330,25 +330,38 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
-size_t ucs_get_huge_page_size()
+int ucs_get_meminfo_entry(char *pattern)
 {
-    static size_t huge_page_size = 0;
     char buf[256];
-    int size_kb;
+    char final_pattern[80];
+    int val = 0;
     FILE *f;
+    f = fopen("/proc/meminfo", "r");
+    if (f != NULL) {
+        strcpy(final_pattern, pattern);
+        strcat(final_pattern, ":      %d kB");
+        while (fgets(buf, sizeof(buf), f)) {
+            if (sscanf(buf, final_pattern, &val) == 1) {
+                break;
+            }
+        }
+        fclose(f);
+    }
+    return val * 1024;
+}
+
+size_t ucs_get_memfree_size()
+{
+    return ucs_get_meminfo_entry("MemFree");
```

Comment:
if could not read /proc/meminfo need to return a default value which is not 0, so tests could still work, for example 640kb
see ucs_get_huge_page_size()